### PR TITLE
Add Fit_transform to pipeline_base and component_graph

### LINKED ISSUE #2665: Add `fit_transform()` to pipelines and component graphs
With https://github.com/alteryx/evalml/pull/2625 merged, we can also add a `fit_transform()` method for pipelines / component graphs--probably sufficient to just call .fit().transform()!

### PR DESCRIPTION
**Pull request by deanpatel2**
### Pull Request Description

Fixes #2665

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/tests/pipeline_tests/test_component_graph.py [hunk 1]
```diff
@@ -491,6 +491,51 @@ def test_fit_correct_inputs(
...
+    mock_imputer_transform.return_value = ones_df
+    mock_ohe_transform.return_value = ones_df
+
+    component_graph.fit_transform(X, y)
+
+    assert mock_label_encoder_transform.call_count == 2
+    assert mock_imputer_transform.call_count == 1
+    assert mock_ohe_transform.call_count == 1
+    assert mock_fit_transform.call_count == 2
+
```
Comment (christopherbunn) [inline evalml/tests/pipeline_tests/test_component_graph.py:537]: Did you ever figure out why we had to mock the encoder call twice and the other transformer components once? Not blocking but just curious to see if you found out.

**Review by christopherbunn — APPROVED** [review]
Excellent, thanks for doing this!

File: evalml/tests/pipeline_tests/test_component_graph.py (hunk 1 above)
Comment (deanpatel2) [reply evalml/tests/pipeline_tests/test_component_graph.py:537]: haven't got a chance to dig deeper there unfortunately

Comment (chukarsten) [reply evalml/tests/pipeline_tests/test_component_graph.py:537]: Would definitely like to know the answer here before moving on!

**Review by chukarsten — CHANGES_REQUESTED** [review]
Think I would just like an answer to Chris' question about the multiple calls being required.

File: evalml/tests/pipeline_tests/test_component_graph.py (hunk 1 above)
Comment (deanpatel2) [reply evalml/tests/pipeline_tests/test_component_graph.py:537]: Found out the `LabelEncoder` class overrides its `fit_transform` method with `self.fit().transform()`, so when the LabelEncoder transform mock is called, it actually triggers 2 calls to its `transform()` unlike the OHE and Imputer

**Comment by deanpatel2** [comment]
Note: Found out the LabelEncoder class overrides its `fit_transform` method with its own `self.fit().transform()`. Thus, when its mock is called, it actually triggers 2 calls to its `transform()` unlike the OHE and Imputer

**Review by jeremyliweishih — APPROVED** [review]
LGTM! great work @deanpatel2

**Pull request merged**